### DISCUSSION TITLE
Alternative exercises in a program

### DISCUSSION THREAD

**Discussion by efaulhaber** [astashov/liftosaur#173] [Ideas]
As an alternative to https://github.com/astashov/liftosaur/discussions/172, one could specify alternative options in a program.
Say, on Day B of my program, I want to do either cable rows or use a seated rowing machine, depending on current availability of machines and current mood.
It would be great if I could write a program maybe like
```
options {
Cable Row / 3x10 80kg
Seated Rowing Machine / 3x10 50kg
}
```
And then during the workout, I have to select one of them.

Currently, what I do is add Cable Row to the program and substitute when I feel like it or when the machine is occupied.
This is a bit tedious, as I first have to search for the correct exercise in the substitution menu, and then I have to remember how much weight I need on this other machine (or check the previous workout logs for the weights I used last time).

**Comment by steshaw** [comment]
Something like this would be good. My wife's PT has given her an alternative exercise for 5 out of 6 exercises so that she can choose for herself on a whim or if the machine is busy. It's a two day "full body" program. At the moment, I've set it up as a two day program with the first alternative on day 1 and the second alternative on day 2. This isn't ideal or accurate. Alternatively, I could add a single day with the first alternative but my wife would need to swap while tracking her workouts and remember the right exercise to switch to. Plus, I imagine that the weights won't be tracked in the same way for swapped exercises...

My wife and I like the app anyhow 😁. Thanks!

**Comment by sebastiansm** [comment]
I was looking for this feature, I only have 50min to workout in the mornings so can't wait for equipment.

Yesterday the leg press machines were occupied so I did leg extensions, but didn't log it as a new exercise or replacement exercise (was short on time). 

It would be useful to have a plan B/plan C exercise and be able to log that progression when necessary.

**Comment by aamaratunga** [comment]
Similar to this idea is having "movement patterns". So instead of specifying an exercise like Leg Press, you can specify "Squat Pattern" and then you get a dropdown of the squat movements like Leg Press, Barbell Squat, Hack Squat, etc

**Comment by pat-acorn** [comment]
Currently doing 5/3/1, and something similar would be really useful for accessory work. Basically make groups of related exercises (for 5/3/1 groups would be push, pull, and leg/core) with a rep goal for the whole group, but tracking progression for each exercise in that group individually.

**Comment by lucatortiglione** [comment]
If i can suggest it, Strengthlog has a very good database of excercises and alternatives to switch to

**Comment by bjornmartensson** [comment]
It would be great if the options could be manipulated via Liftoscript, similarly to Set Variations.

A ! could indicate the starting option, and if no !, then the user is prompted for which option should be used.
For example, 
```
options {
!  Cable Row / 3x10 80kg
   Seated Rowing Machine / 3x10 50kg
} {~
  if (completedReps >= reps) {
    weights = weights[ns] + 5lb
    optionIndex += 1
  } 
~}
```
This would allow programs where you stick with one exercise until you clear a certain number of reps, and then move on to an alternative exercise, e.g. alternating between push-ups and bench press.

**Comment by ege-ah** [comment]
this would be amazing for programs like [WSFSB](https://www.scribd.com/document/192423733/Westside-for-Skinny-Bastards) where you have a selection of exercises in a pool
